backend/kale/common/snapshotutils.py

```diff
@@ -0,0 +1,119 @@
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  To allow a notebook to create Rook CephFS snapshot a ClusterRole
+#  and RoleBinding for the default-editor service account in the
+#  given namespace must be applied. Below are examples for use with the
+#  namespace "admin".
+#
+#  !!!WARNING!!!
+#  Might not be secure, only use for testing
+#
+#  apiVersion: rbac.authorization.k8s.io/v1
+#  kind: ClusterRole
+#  metadata:
+#    name: snapshot-access
+#  rules:
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshots"]
+#      verbs: ["create", "get", "list", "watch", "patch", "delete"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotcontents"]
+#      verbs: ["create", "get", "list", "watch", "update", "delete"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotclasses"]
+#      verbs: ["get", "list", "watch"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotcontents/status"]
+#      verbs: ["update"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshots/status"]
+#      verbs: ["update"]
+#    - apiGroups: ["snapshot.storage.k8s.io"]
+#      resources: ["volumesnapshotclasses"]
+#      verbs: ["get", "list", "watch"]
+#    - apiGroups: ["storage.k8s.io"]
+#      resources: ["storageclasses"]
+#      verbs: ["get", "list"]
+#
+#  apiVersion: rbac.authorization.k8s.io/v1
+#  kind: RoleBinding
+#  metadata:
+#    name: allow-snapshot-nb-admin
+#    namespace: admin
+#  subjects:
+#  - kind: ServiceAccount
+#    name: default-editor
+#    namespace: admin
+#  roleRef:
+#    kind: ClusterRole
+#    name: snapshot-access
+#    apiGroup: rbac.authorization.k8s.io
+
+import logging
+
+from kale.common import podutils, k8sutils
+
+NOTEBOOK_SNAPSHOT_COMMIT_MESSAGE = """\
+This is a snapshot of notebook {} in namespace {}.
+This snapshot was created by Kale in order to clone the volumes of the notebook
+and use them to spawn a Kubeflow pipeline.\
+"""
+
+log = logging.getLogger(__name__)
+
+
+def snapshot_pvc(snapshot_name, pvc_name, image="", path="", **kwargs):
```

Comment:
Why not pass a general `annotations` argument here, instead of restricting the possible annotations to `image` and `path`?

Reply:
That is handled by the `**kwargs`.

Reply:
I specified image and path as those are the most important ones for general functionality.

Reply:
Makes sense. This will also make the expansion of the metadata that is written to the snapshots easier. I'll get to implementing this change ASAP.